test/unit/bookmark_test.rb

```diff
@@ -6,17 +6,22 @@ class BookmarkTest < ActiveSupport::TestCase
   end
 
   test "my bookmarks should not contain private bookmarks" do
-    enable_login do
-      assert_equal Bookmark.my_bookmarks.include?(bookmarks(:two)), false
-    end
+    User.current = users(:one)
```

Comment:
Use the `as_user` helper which takes a block, then it won't leave User.current set.